src/api/source/mvt.js

```diff
@@ -0,0 +1,269 @@
+        }
+    }
+
+    _buildMetadata(mvtMetadata) {
+        return new Metadata([], mvtMetadata.columns, 0, 0, '');
+    }
+
+    _clone(){
+        return new MVT(this._templateURL, this.metadata);
+    }
+
+    bindLayer(addDataframe, removeDataframe, dataLoadedCallback) {
+        this._addDataframe = addDataframe;
+        this._removeDataframe = removeDataframe;
+        this._dataLoadedCallback = dataLoadedCallback;
+    }
+
+    async requestMetadata(viz) {
+        this._MNS = viz.getMinimumNeededSchema();
+        return this.metadata;
+    }
+
+    requestData(viewport) {
+        const tiles = rsys.rTiles(viewport);
+        this._getTiles(tiles);
+    }
+
+    free() {
+        this.cache.reset();
+        this._oldDataframes = [];
+    }
+
+    _getTiles(tiles) {
+        this._requestGroupID++;
+        var completedTiles = [];
+        var needToComplete = tiles.length;
+        const requestGroupID = this._requestGroupID;
+        tiles.forEach(t => {
+            const { x, y, z } = t;
+            this._getDataframe(x, y, z).then(dataframe => {
+                if (dataframe.empty) {
+                    needToComplete--;
+                } else {
+                    completedTiles.push(dataframe);
+                }
+                if (completedTiles.length == needToComplete && requestGroupID == this._requestGroupID) {
+                    this._oldDataframes.map(d => d.active = false);
+                    completedTiles.map(d => d.active = true);
+                    this._oldDataframes = completedTiles;
+                    this._dataLoadedCallback();
+                }
+            });
+        });
+    }
+
+    _getDataframe(x, y, z) {
+        const id = `${x},${y},${z}`;
+        const c = this.cache.get(id);
+        if (c) {
+            return c;
+        }
+        const promise = this._requestDataframe(x, y, z);
+        this.cache.set(id, promise);
+        return promise;
+    }
+
+    _requestDataframe(x, y, z) {
+        const mvt_extent = 4096;
+
+        return fetch(this._getTileUrl(x, y, z))
+            .then(rawData => rawData.arrayBuffer())
+            .then(response => {
+
+                if (response.byteLength == 0 || response == 'null') {
+                    return { empty: true };
+                }
+                var tile = new VectorTile(new Protobuf(response));
+                const mvtLayer = tile.layers[Object.keys(tile.layers)[0]];
+                var fieldMap = {};
+                const numFields = [];
+                const jsonFields = [];
+                const stringFields = [];
+                this.metadata.columns.map(c => {
+                    const type = c.type;
+                    if (type == 'json') {
+                        jsonFields.push(c.name);
+                    } else if (type == 'number') {
+                        numFields.push(c.name);
+                    } else if (type == 'category') {
+                        stringFields.push(c.name);
```

Comment:
Right now this is just plain strings, not categories because we can use string data coming from MVT. We should figure out how to use categories